proto/proxy/destination.proto

```diff
@@ -87,6 +87,7 @@ message WeightedAddr {
 message TlsIdentity {
   oneof strategy {
     K8sPodNamespace k8s_pod_namespace = 1;
+    K8sPodIdentity k8s_pod_identity = 2;
```

Comment:
We could eventually remove `K8sPodNamespace`, once the proxy is updated to stop using that strategy.

Reply:
Yes, I already have a similar, but different, change queued up. I'll rebase it on top of this.